{"schemaVersion":{"patch":0,"major":0,"minor":3},"abstract":[{"text":"BluetoothServiceError lists different errors that can be propagated by ","type":"text"},{"type":"reference","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/BluetoothService","isActive":true},{"type":"text","text":" implementations."}],"topicSections":[{"generated":true,"title":"Enumeration Cases","anchor":"Enumeration-Cases","identifiers":["doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/BluetoothServiceError\/notAvailable(_:)","doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/BluetoothServiceError\/notConnected(_:)"]},{"generated":true,"title":"Default Implementations","anchor":"Default-Implementations","identifiers":["doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/BluetoothServiceError\/Error-Implementations"]}],"variants":[{"traits":[{"interfaceLanguage":"swift"}],"paths":["\/documentation\/vigramsdk\/bluetoothserviceerror"]}],"sections":[],"metadata":{"title":"BluetoothServiceError","role":"symbol","roleHeading":"Enumeration","navigatorTitle":[{"text":"BluetoothServiceError","kind":"identifier"}],"fragments":[{"text":"enum","kind":"keyword"},{"text":" ","kind":"text"},{"text":"BluetoothServiceError","kind":"identifier"}],"externalID":"s:9VigramSDK21BluetoothServiceErrorO","modules":[{"name":"VigramSDK"}],"symbolKind":"enum"},"hierarchy":{"paths":[["doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK"]]},"identifier":{"url":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/BluetoothServiceError","interfaceLanguage":"swift"},"seeAlsoSections":[{"generated":true,"title":"Errors","anchor":"Errors","identifiers":["doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/AuthenticationError","doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/NtripServiceError","doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/NtripTaskError","doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/PeripheralError","doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/SinglePointRecordingError"]}],"kind":"symbol","primaryContentSections":[{"kind":"declarations","declarations":[{"tokens":[{"text":"enum","kind":"keyword"},{"text":" ","kind":"text"},{"text":"BluetoothServiceError","kind":"identifier"}],"platforms":["iOS"],"languages":["swift"]}]}],"relationshipsSections":[{"title":"Conforms To","kind":"relationships","identifiers":["doc:\/\/com.vigram.VigramSDK\/s5ErrorP","doc:\/\/com.vigram.VigramSDK\/s8SendableP"],"type":"conformsTo"}],"references":{"doc://com.vigram.VigramSDK/documentation/VigramSDK/NtripTask":{"kind":"symbol","fragments":[{"kind":"keyword","text":"protocol"},{"kind":"text","text":" "},{"kind":"identifier","text":"NtripTask"}],"title":"NtripTask","abstract":[{"type":"text","text":"A NtripTask requests RTK data over a NTRIP connection."}],"navigatorTitle":[{"kind":"identifier","text":"NtripTask"}],"role":"symbol","type":"topic","url":"\/documentation\/vigramsdk\/ntriptask","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/NtripTask"},"doc://com.vigram.VigramSDK/documentation/VigramSDK/BluetoothServiceError/Error-Implementations":{"title":"Error Implementations","abstract":[],"role":"collectionGroup","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/BluetoothServiceError\/Error-Implementations","kind":"article","type":"topic","url":"\/documentation\/vigramsdk\/bluetoothserviceerror\/error-implementations"},"doc://com.vigram.VigramSDK/documentation/VigramSDK/SinglePointRecordingError":{"role":"symbol","url":"\/documentation\/vigramsdk\/singlepointrecordingerror","fragments":[{"kind":"keyword","text":"enum"},{"text":" ","kind":"text"},{"kind":"identifier","text":"SinglePointRecordingError"}],"type":"topic","title":"SinglePointRecordingError","abstract":[{"type":"text","text":"SinglePointRecordingError are thrown, when due to the filtering phase of single point recordings there is not enough valid data to create a singlePoint."}],"kind":"symbol","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/SinglePointRecordingError","navigatorTitle":[{"kind":"identifier","text":"SinglePointRecordingError"}]},"doc://com.vigram.VigramSDK/documentation/VigramSDK/NtripTaskError":{"kind":"symbol","fragments":[{"text":"enum","kind":"keyword"},{"text":" ","kind":"text"},{"text":"NtripTaskError","kind":"identifier"}],"abstract":[{"type":"text","text":"NtripTaskError lists different errors that can be emitted by "},{"identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/NtripTask","type":"reference","isActive":true},{"type":"text","text":" implementations."}],"title":"NtripTaskError","navigatorTitle":[{"text":"NtripTaskError","kind":"identifier"}],"role":"symbol","type":"topic","url":"\/documentation\/vigramsdk\/ntriptaskerror","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/NtripTaskError"},"doc://com.vigram.VigramSDK/documentation/VigramSDK/BluetoothServiceError/notConnected(_:)":{"title":"BluetoothServiceError.notConnected(_:)","abstract":[{"type":"text","text":"Connecting to a device failed unexplicably."}],"role":"symbol","fragments":[{"text":"case","kind":"keyword"},{"text":" ","kind":"text"},{"text":"notConnected","kind":"identifier"},{"text":"(","kind":"text"},{"text":"UUID","kind":"typeIdentifier","preciseIdentifier":"s:10Foundation4UUIDV"},{"text":")","kind":"text"}],"identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/BluetoothServiceError\/notConnected(_:)","kind":"symbol","type":"topic","url":"\/documentation\/vigramsdk\/bluetoothserviceerror\/notconnected(_:)"},"doc://com.vigram.VigramSDK/documentation/VigramSDK/PeripheralError":{"role":"symbol","url":"\/documentation\/vigramsdk\/peripheralerror","fragments":[{"kind":"keyword","text":"enum"},{"kind":"text","text":" "},{"text":"PeripheralError","kind":"identifier"}],"title":"PeripheralError","abstract":[{"type":"text","text":"PeripheralError combines different failure events that can be emitted on sending data to a "},{"type":"reference","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/Peripheral","isActive":true},{"type":"text","text":"."}],"type":"topic","kind":"symbol","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/PeripheralError","navigatorTitle":[{"text":"PeripheralError","kind":"identifier"}]},"doc://com.vigram.VigramSDK/documentation/VigramSDK":{"role":"collection","images":[{"identifier":"logo.png","type":"icon"}],"url":"\/documentation\/vigramsdk","type":"topic","title":"VigramSDK","abstract":[],"kind":"symbol","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK"},"doc://com.vigram.VigramSDK/documentation/VigramSDK/BluetoothServiceError":{"role":"symbol","url":"\/documentation\/vigramsdk\/bluetoothserviceerror","fragments":[{"text":"enum","kind":"keyword"},{"text":" ","kind":"text"},{"text":"BluetoothServiceError","kind":"identifier"}],"abstract":[{"text":"BluetoothServiceError lists different errors that can be propagated by ","type":"text"},{"isActive":true,"type":"reference","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/BluetoothService"},{"text":" implementations.","type":"text"}],"type":"topic","title":"BluetoothServiceError","kind":"symbol","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/BluetoothServiceError","navigatorTitle":[{"text":"BluetoothServiceError","kind":"identifier"}]},"doc://com.vigram.VigramSDK/s8SendableP":{"identifier":"doc:\/\/com.vigram.VigramSDK\/s8SendableP","title":"Swift.Sendable","type":"unresolvable"},"doc://com.vigram.VigramSDK/documentation/VigramSDK/BluetoothService":{"title":"BluetoothService","abstract":[{"type":"text","text":"A BluetoothService is used to handle bluetooth connections."}],"role":"symbol","navigatorTitle":[{"text":"BluetoothService","kind":"identifier"}],"fragments":[{"text":"protocol","kind":"keyword"},{"text":" ","kind":"text"},{"text":"BluetoothService","kind":"identifier"}],"identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/BluetoothService","kind":"symbol","type":"topic","url":"\/documentation\/vigramsdk\/bluetoothservice"},"doc://com.vigram.VigramSDK/documentation/VigramSDK/AuthenticationError":{"role":"symbol","url":"\/documentation\/vigramsdk\/authenticationerror","fragments":[{"text":"enum","kind":"keyword"},{"text":" ","kind":"text"},{"text":"AuthenticationError","kind":"identifier"}],"title":"AuthenticationError","type":"topic","abstract":[{"type":"text","text":"AuthenticationError contains custom failure events when connecting to Authentication process."}],"identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/AuthenticationError","kind":"symbol","navigatorTitle":[{"text":"AuthenticationError","kind":"identifier"}]},"logo.png":{"type":"image","variants":[{"url":"\/images\/com.vigram.VigramSDK\/logo.png","traits":["1x","light"]}],"identifier":"logo.png","alt":"Vigram logo."},"doc://com.vigram.VigramSDK/documentation/VigramSDK/NtripServiceError":{"role":"symbol","url":"\/documentation\/vigramsdk\/ntripserviceerror","fragments":[{"kind":"keyword","text":"enum"},{"kind":"text","text":" "},{"kind":"identifier","text":"NtripServiceError"}],"type":"topic","abstract":[{"text":"NtripServiceError contains custom failure events when connecting to Ntrip services.","type":"text"}],"title":"NtripServiceError","kind":"symbol","identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/NtripServiceError","navigatorTitle":[{"kind":"identifier","text":"NtripServiceError"}]},"doc://com.vigram.VigramSDK/s5ErrorP":{"type":"unresolvable","identifier":"doc:\/\/com.vigram.VigramSDK\/s5ErrorP","title":"Swift.Error"},"doc://com.vigram.VigramSDK/documentation/VigramSDK/Peripheral":{"title":"Peripheral","abstract":[{"type":"text","text":"Peripheral provides a write and receive interface to communicate with the viDoc device."}],"role":"symbol","navigatorTitle":[{"text":"Peripheral","kind":"identifier"}],"fragments":[{"text":"protocol","kind":"keyword"},{"text":" ","kind":"text"},{"text":"Peripheral","kind":"identifier"}],"identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/Peripheral","kind":"symbol","type":"topic","url":"\/documentation\/vigramsdk\/peripheral"},"doc://com.vigram.VigramSDK/documentation/VigramSDK/BluetoothServiceError/notAvailable(_:)":{"title":"BluetoothServiceError.notAvailable(_:)","abstract":[{"text":"The device to connect to cannot be found, since it is (possibly no longer) available.","type":"text"}],"role":"symbol","fragments":[{"kind":"keyword","text":"case"},{"text":" ","kind":"text"},{"text":"notAvailable","kind":"identifier"},{"kind":"text","text":"("},{"text":"UUID","kind":"typeIdentifier","preciseIdentifier":"s:10Foundation4UUIDV"},{"text":")","kind":"text"}],"identifier":"doc:\/\/com.vigram.VigramSDK\/documentation\/VigramSDK\/BluetoothServiceError\/notAvailable(_:)","kind":"symbol","type":"topic","url":"\/documentation\/vigramsdk\/bluetoothserviceerror\/notavailable(_:)"}}}